Add flair to snackbar

miikkatu opened the pull request:
This is for #1279.

<img width="786" alt="screen shot 2018-04-10 at 14 06 56" src="https://user-images.githubusercontent.com/16205520/38554270-a3e00986-3cca-11e8-8a1d-ceef26e053d6.png">

Unlike modal dialogs in the redesign, this is with a dark background to make it pop a bit more. Functionality remains the same as with the original snackbar.

The prop-types package was added as a dependency.

seanyesmunt commented:
Nice. I will check out the code more later but the picture looks good. We are using `flow` instead of `prop-types` so your types should be moved to that.

seanyesmunt commented on src/renderer/scss/component/_snack-bar.scss:
```diff
@@ -1,30 +1,54 @@
+  background-color: var(--color-primary);
+  border-radius: 10px;
+  box-shadow: var(--box-shadow-layer);
+  color: #f0f0f0;
```
This should use a text color variable. Probably just `--text-color` or whatever we call it.

miikkatu replied:
There's a snackbar specific variable now.

seanyesmunt commented on src/renderer/scss/component/_snack-bar.scss:
```diff
@@ -1,30 +1,54 @@
-  padding: $spacing-vertical;
-  position: fixed;
-  top: $spacing-vertical;
+  background-color: var(--color-primary);
```
You should create a variable for this `--snack-bar-bg` `--snack-bar-color`

So we can change them for dark mode. Similar to how we do it with buttons:
https://github.com/lbryio/lbry-app/blob/master/src/renderer/scss/_vars.scss#L93

seanyesmunt commented on src/renderer/scss/component/_snack-bar.scss:
```diff
@@ -1,30 +1,54 @@
+    text-transform: uppercase;
+
+    span {
+      color: #f0f0f0;
```
This should just be a color variable. It's just the same as the color you have above so maybe this isn't needed?

miikkatu commented:
@seanyesmunt I updated the code as suggested in the comments, which consequently were cleared because I updated the commit and force pushed here again. 😅

seanyesmunt commented on src/renderer/scss/component/_snack-bar.scss:
```diff
@@ -1,30 +1,51 @@
-  min-width: min-content;
-  &:hover {
-    text-decoration: underline;
+  .message {
```
Looks good. The only thing I would do is change this to `snack-bar__message` and move it to the top level. Then I think it's ok to merge.

miikkatu commented:
PR changed per review. Also, `snack-bar__action` was moved to the top level along with `snack-bar__message`.

seanyesmunt commented:
Awesome! Just add a changelog commit and I will merge

miikkatu commented:
Changelog updated.

tzarebczan commented:
Thanks @miikkatu! Have your contribution email, will follow up soon!
